gradio/documentation.py

```diff
@@ -177,3 +176,16 @@ def generate_documentation():
                 )
             documentation[mode].append(cls_documentation)
     return documentation
+
+
+def document_component_api(component_cls, target):
```

Comment:
Can you give some explanation, what's happening here?

Reply:
Added documentation. Uses